BlogTemplate/Pages/Account/ConfirmEmail.cshtml.cs

```diff
@@ -28,13 +28,13 @@ public ConfirmEmailModel(UserManager<ApplicationUser> userManager)
             var user = await _userManager.FindByIdAsync(userId);
             if (user == null)
             {
-                throw new ApplicationException($"Unable to load user with ID '{userId}'.");
+                throw new FieldAccessException($"Unable to load user with ID '{userId}'.");

```

Comment:
We've found that all of the files whose exceptions need to be changed require that the user be logged in to access the page. Because of this we will never run into a user not found error. In any Get request where we need to edit the exception we think of using the InvalidOperationException and on a post request we will use a BadRequestException.